New Workspace RBAC changes (#2737)
astronomer/docs: astro/user-permissions.md

- A **Workspace Owner** has all the same permissions as a Workspace Operator, plus the ability to manage user membership in the Workspace. Give a user this role if they need to administrate membership to the Workspace. 

The following table lists the specific permissions that each Workspace role has:

| Permission                                                                  | **Workspace Member** | **Workspace Author** | **Workspace Operator** | **Workspace Owner** |
| --------------------------------------------------------------------------- | -------------------- | -------------------- | ---------------------- | ------------------- |
| View Workspace users                                                        | ✔️                    | ✔️                    | ✔️                      | ✔️                   |
| View all Deployments in the Cloud UI                                        | ✔️                    | ✔️                    | ✔️                      | ✔️                   |
| View DAGs in the Airflow UI                                                 | ✔️                    | ✔️                    | ✔️                      | ✔️                   |

> | View Airflow task logs                                                      | ✔️                    | ✔️                    | ✔️                      | ✔️                   |

Author: ```suggestion
| View Airflow task logs                                                      | ✔️                    | ✔️                    | ✔️                      | ✔️                   |
| View Airflow datasets                                                       | ✔️                    | ✔️                    | ✔️                      | ✔️                   |
```

| Create, update, and delete Astro Cloud IDE projects                         |                      | ✔️                    | ✔️                      | ✔️                   |
| Create, update, and delete Astro alerts                                     |                      | ✔️                    | ✔️                      | ✔️                   |
| View Airflow connections, variables, plugins, providers, pools, and XComs   |                      | ✔️                    | ✔️                      | ✔️                   |
| Update Airflow connections, variables, plugins, providers, pools, and XComs |                      |                      | ✔️                      | ✔️                   |
| Update Deployment configurations                                            |                      |                      | ✔️                      | ✔️                   |
| Create and Delete Deployments                                               |                      |                      | ✔️                      | ✔️                   |

> | Create, Update and Delete Environment Variables                             |                      |                      | ✔️                      | ✔️                   |

Reviewer: ```suggestion
| Create, Update and Delete Deployment Environment Variables                             |                      |                      | ✔️                      | ✔️                   |
```

> | Create, Update and Delete Deployment Environment Variables                             |                      |                      | ✔️                      | ✔️                   |

Author: ```suggestion
| Create, Update and Delete Deployment Environment Variables                  |                      |                      | ✔️                      | ✔️                   |
```

## Workspace roles

A Workspace role grants a user or API token some level of access to a specific Workspace. If a user or API token has some level of access to a Workspace, that access applies to all Deployments in the Workspace. 

> - A **Workspace Member** has the least permissions in a Workspace and can only view the most basic details about Deployments and DAGs. Give a user this role if X

Reviewer 2: Looks like some stand-in text didn't get fully updated. Took a stab at it. 
```suggestion
- A **Workspace Member** has the least permissions in a Workspace and can only view the most basic details about Deployments and DAGs. Give a user this role if they need to be able to view a DAG's run status or Deployment health.
```

| Create, update, and delete Astro Cloud IDE projects                         |                      | ✔️                    | ✔️                      | ✔️                   |
| Create, update, and delete Astro alerts                                     |                      | ✔️                    | ✔️                      | ✔️                   |
| View Airflow connections, variables, plugins, providers, pools, and XComs   |                      | ✔️                    | ✔️                      | ✔️                   |
| Update Airflow connections, variables, plugins, providers, pools, and XComs |                      |                      | ✔️                      | ✔️                   |
| Update Deployment configurations                                            |                      |                      | ✔️                      | ✔️                   |
| Create and Delete Deployments                                               |                      |                      | ✔️                      | ✔️                   |

> | Create, Update and Delete Deployment environment variables                  |                      |                      | ✔️                      | ✔️                   |

Author: ```suggestion
| Create, Update and Delete Deployment environment variables                  |                      |                      | ✔️                      | ✔️                   |
| View the **Cluster Activity** tab in the Airflow UI                  |                      |                      | ✔️                      | ✔️                   |
```

| Create, update, and delete Astro alerts                                     |                      | ✔️                    | ✔️                      | ✔️                   |
| View Airflow connections, variables, plugins, providers, pools, and XComs   |                      | ✔️                    | ✔️                      | ✔️                   |
| Update Airflow connections, variables, plugins, providers, pools, and XComs |                      |                      | ✔️                      | ✔️                   |
| Update Deployment configurations                                            |                      |                      | ✔️                      | ✔️                   |
| Create and Delete Deployments                                               |                      |                      | ✔️                      | ✔️                   |
| Create, Update and Delete Deployment environment variables                  |                      |                      | ✔️                      | ✔️                   |

> | View the **Cluster Activity** tab in the Airflow UI                  |                      |                      | ✔️                      | ✔️                   |

Author: ```suggestion
| View the **Cluster Activity** tab in the Airflow UI                       |                      |                      | ✔️                      | ✔️                   |
```

> | View the **Cluster Activity** tab in the Airflow UI                       |                      |                      | ✔️                      | ✔️                   |

Author: ```suggestion
| View the **Cluster Activity** tab in the Airflow UI                         |                      |                      | ✔️                      | ✔️                   |
```

> | View the **Cluster Activity** tab in the Airflow UI                         |                      |                      | ✔️                      | ✔️                   |

Author: ```suggestion
| View the **Cluster Activity** tab in the Airflow UI                         |                      |                      |                        | ✔️                   |
```
